### PR TITLE
Railgun nerf

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Basically reverts railgun back to being a single shotgun.

Decreased railgun pierce mult from 0.85 to 0.75 (this reduces dam and sunder when piercing anything, such as mobs/turfs).
Added a range reduction when going through turfs by 3 per turf, so you can't shoot xenos through 10 walls.

Edit 2: Also fixes the exploit of removing all the bullets out of railgun mags which let you store huge amounts of ammo for very little space. It now behaves like rocket launchers, where the projectile stays with the magazine always.
## Why It's Good For The Game
Its like a sidegrade to sadar or at least RR, can pen anything so can be shot through friendlies/walls, moves extremely fast so is effectively impossible to dodge, on top of having no windup unlike rockets.

Its also much cheaper.
## Changelog
:cl:
balance: Railgun changed back to 1 shot instead of 3
balance: Reduced railgun effectiveness when piercing things
fix: Fixed being able to remove railgun rounds of their magazine
/:cl:

### CODE REVIEW

**Comment by Technical-l** [comment]
yo wait wasn't the railgun rebalanced around having 3 shots or am i trippin? like i swear the stats were nerfed because the capacity was increased

**Comment by Araulius** [comment]
> yo wait wasn't the railgun rebalanced around having 3 shots or am i trippin? like i swear the stats were nerfed because the capacity was increased

you are correct, it was. after it had 100 sunder damage removed.

**Comment by GravitronBlue** [comment]
Since my apperant xope managed to inspire this PR I was planning add some feedback, yet after reviewing the PR history of the railgun it seems there's a bit more to why it's so effective right now.

#### Short answer?/TLDR
A cost efficient, heavy sunder (In the form of HVAP mags), wall piercing weapon that both cheap to buy, cheap to sustain, and unga-tier easy to hit with, and capable of crowd control against xeno group pushes.

Generally takes a miracle/marines throwing to save the hive once the HVAP spam starts and the whole hive is locked down by not having armor to push with.

#### Long answer?/Story
At just 400 points for the gun and 50 per mag type, marines getting a couple of these guns for free via seasonals and buying a few HVAP mags can make it possible to consistently rush this weapon at roundstart.

Firepower wise you get the most attrition heavy damage type against xenos in the game, _**sunder**_.
How do you apply it?
By shooting them **_through walls_**.

This merge in concept is **_HORRIBLE_**. (Like with APCR on TAT, only this is a handheld version)
Nevermind that you can afford this roundstart/during prep phase.

Were this somehow a xeno equivalant imagine warlock having psy lance the moment shutters open, being able to shoot it through walls, dealing cloneloss, and having no chargeup time. (Maybe next year on April Fools)
That's all this minus having xray vision, but who needs that when you can just snipe with NVGs and helmet binoculars?

ScopedNVG
Range: 21 tiles/3 screens apperantly too minus some per wallpen, basically a half-ish range AMR that
can shoot through a few walls, Nice!

Not a real surprise given the plagued history of this weapon such as the time when marines would juggle this railgun for the instant 100 sunder it once did (back before it was given wallpiercing from what I can recall).

As for the PR here?
Doesn't really change much other than requiring a reload between the weapon's normal cooldown between shots and slight cost increase, a bit more difficult to carry ammo but really just means lugging around an ammobox to restock from.

~~I'm blaming most of this on #10139 and #Walance for _resolute_ powercreep.~~
(Nevermind I forgot about #13020 getting 21 downvotes for bringing railgun sunder to sane levels. Unga brigading moment?)

#### Solutions?
A proper breakdown of the ways this weapon being effective yet low cost should really be listed and covered by #Walance.

Personally I've thought about tackling the weapon's ease of use rather than statchanges by making it dependant on charging up first before firing (Helldivers 2 style), along with allowing warlocks to properly catch and reflect it. (Why they can't to begin with would be news to me as well)
HVAP mags being the same price as everything else despite them being generally what's best for the weapon is also questionable.

Thanks anyway I suppose for the PR and attention towards the state of railgun balance. (Doubt it'll be the last though)
>_"If you lost the game you played the game wrong..."?_ Truth is, you didn't rig it from the start.

**Comment by Lauwma** [comment]
1 shot per 50 points is very expensive, at least cut it in half. 3 shots of railgun don't have the same value as an armor module, combat backpack, a tangle OB. That means you'd get 1 shot of railgun every minute per miner. Unlike the SADAR railgun aren't a guaranteed death sentence for the xenos, it's just a cue to back off. Even when I play T1s I don't die from a railgun, it just makes me back off and delay my attack for 15 seconds later while I heal sunder and health.

**Comment by Lumipharon** [comment]
well you're in luck, the pr already changes the mag cost.

**Comment by GravitronBlue** [comment]
> 1 shot per 50 points is very expensive

Cheaper than an AP sadar rocket (nevermind the price for buying the launcher) and you can fire it through walls, raw kill potential isn't what makes this gun good, it's how easilly you can apply sunder to anything that's visible to you with absolutely zero counterplay,
-walls don't work, (Not even special walltypes)
-warlock shield doesn't work
-gorger ought to do more here but overheal won't save sundered xenos and gorgers themselves don't have kill potential,
-hivelords will never keep up with the amount of sunder you can constantly dish out nor can they fix this level of sunder mid-combat where having armor matters.
-defender dies the moment it openly challenges this thing (stagger+sunder) and will stay stuck waiting on its regrow carapace ability cooldown.

but if you really think it should be cheaper... how about 10 points per mag?

Oh wait, it **_already is_**.

https://github.com/tgstation/TerraGov-Marine-Corps/blob/0461e7d92d253584778700aa2bf22ef143594c29/code/modules/reqs/supplypacks.dm#L2285-L2288

https://github.com/tgstation/TerraGov-Marine-Corps/blob/0461e7d92d253584778700aa2bf22ef143594c29/code/modules/factory/unboxer.dm#L315-L319
That's 20 mags at a price at a price just under getting 2 armor modules, assuming you had roundstart seasonals and a half competent RO you could have them done via reqtorio by the time prep ends.




> Unlike the SADAR railgun aren't a guaranteed death sentence for the xenos, it's just a cue to back off. Even when I play T1s I don't die from a railgun, it just makes me back off and delay my attack for 15 seconds later while I heal sunder and health.

Except sunder doesn't heal in 15 seconds even with resting weeds, closer to 20 if there's an acid pool available, even with a pocket drone I've found myself stuck in sunder limbo where you can't attack because of heavy sunder and any attempt to pressure the railgun holder afterwards results in being driven back loaded with sunder all the same.

Despite hivelords, drone link, sunderfruit, and armor restoring acid pools; none really help you in active combat and none offer resistance to sunder where it matters, in combat.

In a nutshell:
AP rockets have kill potential if you can get lucky and land a shot.
HVAP railgun gives kill potential to everyone with no effort as you can't miss outside of skill issue.

Short of playing marine myself I've got little idea how to actually convince those here or on #Walance by simply discussing it.

**Comment by Lauwma** [comment]
> > 1 shot per 50 points is very expensive
> 
> 
> 
> Cheaper than an AP sadar rocket (nevermind the price for buying the launcher) and you can fire it through walls, raw kill potential isn't what makes this gun good, it's how easilly you can apply sunder to anything that's visible to you with absolutely zero counterplay,

> -walls don't work, (Not even special walltypes)
> 
> -warlock shield doesn't work
> 
> -gorger ought to do more here but overheal won't save sundered xenos and gorgers themselves don't have kill potential,

It litteraly just halves your armor. The damages are minimal if you were at full damage. Its point is indeed to reduce the survivability of the frontline xenos like crusher, king, warrior and such. Even if you get shot through wall unless you greatly overcommitted and escaped with a pixel of health you are not gonna die. At best you are gonna go crit and crit heal and it's fine because you are behind a wall that covers you. The problem with current railgun is that it can shoot you three times in one fight. If it was single shot the amount of sunder it inflicted would be fine given its reload is pretty long. Also gorger are a support caste and I don't think having a support T3 really is a downside given the enabling it gives to fragile powerful castes like defiler. 

> -hivelords will never keep up with the amount of sunder you can constantly dish out nor can they fix this level of sunder mid-combat where having armor matters.

That's a straight up lie. You can regen 50% sunder in less than 15 seconds of laying down in rest weeds with hivelord beam.

> -defender dies the moment it openly challenges this thing (stagger+sunder) and will stay stuck waiting on its regrow carapace ability cooldown.

Isn't defender stagger immune when they fortify ?

> but if you really think it should be cheaper... how about 10 points per mag?
> 
> 
> 
> Oh wait, it **_already is_**.
> 
> 
> 
> https://github.com/tgstation/TerraGov-Marine-Corps/blob/0461e7d92d253584778700aa2bf22ef143594c29/code/modules/reqs/supplypacks.dm#L2285-L2288
> 
> 
> 
> https://github.com/tgstation/TerraGov-Marine-Corps/blob/0461e7d92d253584778700aa2bf22ef143594c29/code/modules/factory/unboxer.dm#L315-L319
> 
> That's 20 mags at a price at a price just under getting 2 armor modules, assuming you had roundstart seasonals and a half competent RO you could have them done via reqtorio by the time prep ends.

20 shots for 2 armor is still not worth it lol. 60 shots was stupid though 

> > Unlike the SADAR railgun aren't a guaranteed death sentence for the xenos, it's just a cue to back off. Even when I play T1s I don't die from a railgun, it just makes me back off and delay my attack for 15 seconds later while I heal sunder and health.

> Despite hivelords, drone link, sunderfruit, and armor restoring acid pools; none really help you in active combat and none offer resistance to sunder where it matters, in combat.

The point of armor is to be regenerated out of combat. It prevents xenos from just shrugging an infinite amount of damage. 

> In a nutshell:
> 
> AP rockets have kill potential if you can get lucky and land a shot.
> 
> HVAP railgun gives kill potential to everyone with no effort as you can't miss outside of skill issue.

I was more thinking of HE rocket since it's just better than AP. It sunders, brazils, and slows xenos. AP SADAR only exists to nuke T1 and T2, it's actually pretty bad on high armor targets and I would indeed choose a railgun any day over AP sadar against a crusher because one is designed against tanks and another against squishy xenos. 

So yeah I still believe 50 points a shot is stupid. 25 or 30 would be more reasonable

**Comment by Lumipharon** [comment]
why are you nerds arguing about the shots costing 50, the pr already changed them to 30.

**Comment by GravitronBlue** [comment]
Did a couple local tests as well as a trial run in a normal round, no free seasonal, asked RO (Ace) for a 600 point req order of one railgun plus one reqtorio fill order of HVAP. (20 mags), no SG goggles, just a bauldur module (flashlight module) with heavy armor as a droid.

Can't fit it all here unless I make a compile of every railgun shot (Which I really should), but most of my time was spent just taking pot shots at the biggest bug near the frontline until they leave while the ungaball moves forward uncontested.

Wasn't entirely uneventful though so...
#### Here's a couple highlights:

_Rescue push suffers, loses crusher (Zeethh death so does it really count?)_

https://github.com/tgstation/TerraGov-Marine-Corps/assets/17460297/c0dc2d22-e45e-4a80-b4e6-f36f038ff8ee

_Barnet tests frontline fortify and nearly dies (Possibly baiting a railgun shot but just barely escapes)_

https://github.com/tgstation/TerraGov-Marine-Corps/assets/17460297/2fadf760-9f2b-4044-8c8c-31357d221c66

#### Tested locally about 50 sunder recovery times:

20 seconds for 40 sunder recovery.

https://github.com/tgstation/TerraGov-Marine-Corps/assets/17460297/0f480882-0666-475f-8be5-025ebf9092e4

50 sunder to none in 25 seconds

https://github.com/tgstation/TerraGov-Marine-Corps/assets/17460297/4143a000-a0ab-4423-a8de-a398d130e81d

>That's a straight up lie. You can regen 50% sunder in less than 15 seconds of laying down in rest weeds with hivelord beam.

Sus

>why are you nerds arguing about the shots costing 50

Technically just 10 points via reqtorio (20% normal price and surely not price powercreeped)
I'm still rather against misc statchanges like all this, but that's for maintainers to decide I guess.

**Comment by Lumipharon** [comment]
factorio is cringe because it encourages marines to spam anything that is abusable on the cheap, but thats out of scope of this pr.

**Comment by Technical-l** [comment]
> #### Solutions?
> A proper breakdown of the ways this weapon being effective yet low cost should really be listed and covered by #Walance.
> 
> Personally I've thought about tackling the weapon's ease of use rather than statchanges by making it dependant on charging up first before firing (Helldivers 2 style), along with allowing warlocks to properly catch and reflect it. (Why they can't to begin with would be news to me as well) HVAP mags being the same price as everything else despite them being generally what's best for the weapon is also questionable.
> 

learn to code lol

**Comment by ivanmixo** [comment]
> well you're in luck, the pr already changes the mag cost.

Maybe you should put that somewhere in the PR description, you only listed the nerfs

**Comment by chizzy376** [comment]
> well you're in luck, the pr already changes the mag cost.

Could you also please change the cost of the reqtorio counterpart for the ammo to reflect this new cut in cost? Thank you for the PR by the way, railguns are very oppressive right now!

**Comment by Naaanii** [comment]
> factorio is cringe because it encourages marines to spam anything that is abusable on the cheap, but thats out of scope of this pr.

I take pride in this.

File: code/modules/projectiles/guns/specialist.dm [hunk 1]
```diff
@@ -1004,7 +1004,7 @@ Note that this means that snipers will have a slowdown of 3, due to the scope
 	gun_features_flags = GUN_WIELDED_FIRING_ONLY|GUN_WIELDED_STABLE_FIRING_ONLY|GUN_AMMO_COUNTER
 	reciever_flags = AMMO_RECIEVER_MAGAZINES|AMMO_RECIEVER_AUTO_EJECT|AMMO_RECIEVER_CYCLE_ONLY_BEFORE_FIRE
 
-	fire_delay = 3 SECONDS
+	fire_delay = 2 SECONDS
```
Comment (Naaanii) [inline code/modules/projectiles/guns/specialist.dm:1007]: Shouldn't fire delay be removed for being a one shot wonder?

**Comment by Isy232** [comment]
If the sunder spam is too oppressive for xenos, preventing them from fighting effectively, I'd suggest nerfing the sunder instead.

Excluding the sunder I've heard very few complaints regarding the railgun and this seems to be going too far.

**Comment by Looking4bros** [comment]
Just use a long SHATTER effect or something lmao. Like the Laser sniper, Squashhead SMG rounds, or Tricordrazine Vali. The effect is very good, but the weapons that have it aren't good at following up on it.It's a temporary 0.2 multiplier on the armor values.

**Comment by Isy232** [comment]
All of these changes combined nerf the railgun far too much.

Mag cost 50 => 30
Mag capacity 3 => 1

Old Ammo Price:
50/3 = 16.66 a shot
New Ammo Price:
30/1 = 30 a shot

Due to only 1 shot in the same size magazine, the amount of railgun ammo possible to carry is cut to a third.

In all:
Railgun only able to fire 1 shot per typical combat encounter.
Railgun ammo costs almost 2x per shot.
Railgun user only able to carry 1/3rd the ammo they previously could.

**Comment by Arturlang** [comment]
Currently railgun a funny click to die button if you're fighting, especially when it can go through walls and deals nutty amounts of damage, comparable to a shotgun. Atleast with a shotgun you can outrange it. Railgun? Much more damaging range

**Comment by Isy232** [comment]
TM this for like 1-2 weeks perhaps?

I think the nerf is going way too far, but whatever. TM it and lets see if railgun keeps being used or ends up in trash tier.
